drivers/iscsilib.py

```diff
@@ -303,6 +303,14 @@ def logout(portal, target, all=False):
     except:
         raise xs_errors.XenError('ISCSILogout')
 
+def delete(target):
+    cmd = ["iscsiadm", "-m", "node", "-o", "delete", "-T", target]
+    failuremessage = "Failed to delete target"
+    try:
+        (stdout,stderr) = exn_on_failure(cmd,failuremessage)
+    except:
+        raise xs_errors.XenError('ISCSIDelete')
```

Comment:
As this is trying to clean out a cache and is only(?) an issue on migration, should the delete failing always be considered fatal to the detach operation? I guess the alternative would be to log the failure - but I'm just a little nervous that there might be other scenarios where perhaps the delete does fail and we're now making detach fail in that case?

Reply:
I think with a small refactoring we can make this entirely the problem of RawISCSISR and be 100% confident that it won't impact the code used by other SR types.

Reply:
I'm not nervous about other SR types - I think the current code is perfectly safe for LVMoISCSI and friends.  My concern is around RawISCSI use cases that don't involve migration and whether a failed delete should always be fatal